### ISSUE TITLE
Counter Culture is limiting count update ONCE per COMMIT

### ISSUE THREAD

**Issue by devmtnaing** [magnusvk/counter_culture#221]
As I noticed here, https://github.com/magnusvk/counter_culture/blob/master/lib/counter_culture/extensions.rb#L92

counter_culture is limiting One count update per commit, so 

```
ActiveRecord::Base.transaction do
   model = Model.last
   model.state = "issued"
   model.save

   # Business Logic
   # More Business Logic

   model.state = "redeemed"
   model.save
end
```

Based on above code, We will get `issued_count` incremented, `redeem_count` ignored. 

Any suggestion on how to get the count updated as the model get saved during the single transaction commit?

**Comment by magnusvk** [comment]
@devmtnaing good point. I added a test and fix for this in #222. This logic is from a time where we were running counter updates after commit by default, which has not been the case for a while. In fact, it's not even tested; I spent a little time trying to add tests for that but it's a pain and there's no real reason for it, so I figure why not deprecate it instead.

Can you confirm this fixes your issue?

**Comment by devmtnaing** [comment]
Yes,  I think #222 fixes the issue. Thanks. 👍 

As quick fix for ourselves, we called `self.instance_variable_set('@_counter_culture_active', false)` as `before_save` on the Model iteself.

**Comment by magnusvk** [comment]
@devmtnaing thanks for reporting this; released as 2.0.0 as this is a breaking change for some people.
